app/scripts/migrations/113.ts

```diff
@@ -0,0 +1,64 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 113;
+
+/**
+ * Migrates users that have opensea enabled to blockaid
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  if (!hasProperty(state, 'PreferencesController')) {
+    return state;
+  }
+
+  if (!isObject(state.PreferencesController)) {
+    global.sentry?.captureException?.(
+      new Error(
+        `state.PreferencesController is type: ${typeof state.PreferencesController}`,
+      ),
+    );
+    state.PreferencesController = {};
+  } else if (
+    !hasProperty(state.PreferencesController, 'transactionSecurityCheckEnabled')
+  ) {
+    global.sentry?.captureException?.(
+      new Error(
+        `state.PreferencesController.transactionSecurityCheckEnabled is missing from PreferencesController state`,
+      ),
+    );
+  } else if (
+    typeof state.PreferencesController.transactionSecurityCheckEnabled !==
+    'boolean'
+  ) {
+    global.sentry?.captureException?.(
+      new Error(
+        `state.PreferencesController.transactionSecurityCheckEnabled is type: ${typeof state
+          .PreferencesController.transactionSecurityCheckEnabled}`,
+      ),
+    );
+  }
+
+  delete state.PreferencesController.transactionSecurityCheckEnabled;
```

Comment:
in the PR description
> With Blockaid successfully rolled to users by default and across multiple networks, we will be deprecating the earlier Opensea security alert feature and migrating users to Blockaid.

when we mention we want to migrate users to Blockaid, does this mean we'd also like to set `securityAlertsEnabled` to true if `transactionSecurityCheck` was true?

Reply:
Yep. You are right. We should be setting `securityAlertsEnabled` to true only if `transactionSecurityCheck` was true. Updated [here](https://github.com/MetaMask/metamask-extension/pull/23460/commits/b2e69112b2b2b93b6c0c98399e827e5aa4a51df6)